src/runtime/hexagon/target/fastrpc/include/tvm_remote.idl

```diff
@@ -0,0 +1,51 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+/*
+ * IDL to offload TVM kernels to Hexagon from APPS for multi-domains.
+ */
+#include "remote.idl"
+#include "AEEStdDef.idl"
+
+interface tvm_remote : remote_handle64 {
+   typedef sequence<octet> buffer;
+   typedef unsigned long handle_t;
+
+   long load_library(in sequence<char> soname,
+                     rout handle_t mod_ptr);
+   long get_symbol(in handle_t mod,
+                   in sequence<char> name,
+                   rout handle_t sym_ptr);
+   long kernel(in handle_t mod,
+               in handle_t symbol,
+               inrout sequence <long> scalar,
+               inrout sequence <long> stack,
```

Comment:
Do we need to `rout` to pass the value back to CPU? I assume we only need `in` here. The generated type will be changed from `int*` to `const int*` .

Reply:
Perhaps.  The stub/skel code generated from IDL will do different things for `in`, `rout`, and `inrout` parameters.  This has to do with cache/memory synchronization that happens under the covers.  It may work with just `in`, but we have never tried it.  I'm somewhat reluctant to changing it without more testing.